src/main.cpp

```diff
@@ -408,9 +433,15 @@ int RunnerSubroutine(const CLI_v2::Results& options)
             PBLOG_DEBUG << "CLR : " << clrRecord.FullName();
             clrRecords.emplace_back(clrRecord);
             GetNextRecord();
+            // TODO this is required because the do while loop has a known bug:
+            //   If the query bam contains only one hole, the loop goes inf. The by-strand case is one hole.
+            if (settings.byStrand) {
+                if (clrRecords.size() > 1) break;
```

Comment:
```suggestion
                if (clrRecords.size() > 1) {
                    break;
                }
```

Okay, but I'll fix it properly after this PR